### DISCUSSION TITLE
R.function 的输入Tensor的定义对应源码的什么class

### DISCUSSION THREAD

**Discussion by yejingxin** [mlc-ai/mlc-zh#124] [Q&A]
```
@R.function
  def main(x: Tensor((1, 784), "float32"),
           w0: Tensor((128, 784), "float32"),
           b0: Tensor((128,), "float32"),
           w1: Tensor((10, 128), "float32"),
           b1: Tensor((10,), "float32")):
  pass
```

这里用的是Tensor， 没有用T.Buffer, 我的理解是对比Buffer是一个reference，这里应该Tensor应该是一个const reference，但Tensor没有导入，是tvm.te.Tensor吗？

**Comment by Hzfengsy** [comment]
这里的表达是图层的IR表达 （Relax IR) `@R.function` 而不是 `@T.prim_func`，在Relax IR中我们使用Tensor（其实是R.Tensor），与low-level的Buffer做区分

**Reply by yejingxin** [reply]
那为什么把type hint从`Tensor((1, 784), "float32")`改成`R.Tensor((1, 784), "float32")` 或者`R.Tensor[(1, 784), "float32"]` 都报错呢？是有什么bug吗？

**Reply by Hzfengsy** [reply]
这是现在parser的一个实现问题，暂时还是写成`Tensor`，但知道实际上表示的是`R.Tensor`即可。后续会进行更新
